streams/dshr_strdata_mod.F90

```diff
@@ -1261,7 +1261,7 @@ subroutine shr_strdata_readLBUB(sdat, ns, mDate, mSec, newData, istr, rc)
     call ESMF_TraceRegionExit(trim(istr)//'_setup')
 
     ! if model current date is outside of model lower or upper bound - find the stream bounds
-    if (rDateM < rDateLB .or. rDateM >= rDateUB) then
+    if (rDateM < rDateLB .or. rDateM > rDateUB) then
```

Comment:
I am wondering if you could back out this change - but keep the new solar zenith interpolation and determine if this fixes the restart problem. It would be good to sort this out - given that you already have a sandbox. But that said - I will accept this PR.

Reply:
Well, it looks like you're right! Someone should double check this, because I had a few different sandboxes going and it's possible I made a mistake in the setup, but (if I did it right) I ran test `ERP_Ly3_P72x2.f10_f10_mg37.IHistClm50BgcCrop.cheyenne_intel.clm-cropMonthOutput` using cdeps0.12.39 but with this one change:

```diff
diff --git a/streams/dshr_strdata_mod.F90 b/streams/dshr_strdata_mod.F90
index d457dc8..ce605c3 100644
--- a/streams/dshr_strdata_mod.F90
+++ b/streams/dshr_strdata_mod.F90
@@ -1261,7 +1261,7 @@ contains
     call ESMF_TraceRegionExit(trim(istr)//'_setup')
 
     ! if model current date is outside of model lower or upper bound - find the stream bounds
-    if (rDateM < rDateLB .or. rDateM >= rDateUB) then
+    if (rDateM < rDateLB .or. rDateM > rDateUB) then
        call ESMF_TraceRegionEnter(trim(istr)//'_fbound')
        call shr_stream_findBounds(stream, mDate, mSec,  sdat%mainproc, &
             sdat%pstrm(ns)%ymdLB, dDateLB, sdat%pstrm(ns)%todLB, n_lb, filename_lb, &
```

and that test now passes.

Reply:
Actually, I can double-check my own work... just give me a bit: I'll check out a fresh copy and run all of the previously-failing restart tests to verify that they now pass.

Reply:
Actually, reading back through the history of this, I think I'd like to leave this to someone else to do some further analysis and testing: based on discussion in https://github.com/ESMCI/cime/issues/3380 and https://github.com/ESCOMP/CDEPS/issues/120 it sounds like these two changes are connected. I haven't tried to understand this myself, but maybe @swensosc can comment further.

Reply:
@billsacks - thanks for running the test and confirming my hypothesis. This has nothing to do I believe with the new zenith interpolation but with the test to see if the model current date is outside of the model lower and upper bound.
@swensosc - I believe that you found this problem a while ago. Introducing this fix now seems to break restarts. Not sure what case you ran that needed this fix. Can we find a time to chat about this?